Jenkinsfile

```diff
@@ -0,0 +1,215 @@
+    issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
+  }
+  stages {
+    stage('Initializing'){
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+      }
+      stages {
+        /**
+        Checkout the code and stash it, to use it on other stages.
+        */
+        stage('Checkout') {
+          steps {
+            deleteDir()
+            gitCheckout(basedir: "${BASE_DIR}")
+            stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+          }
+        }
+        /**
+        Lint the code.
+        */
+        stage('Lint') {
+          steps {
+            deleteDir()
+            unstash 'source'
+            script{
+              docker.image('node:8').inside(){
+                dir("${BASE_DIR}"){
+                  sh(label: "Lint", script: 'HOME=$(pwd) .ci/scripts/lint.sh')
+                }
+                stash allowEmpty: true, name: 'cache', includes: "${BASE_DIR}/.npm/**", useDefaultExcludes: false
+              }
+            }
+          }
+        }
+        /**
+        Execute unit tests.
+        */
+        stage('Test') {
+          steps {
+            deleteDir()
+            unstash 'source'
+            dir("${BASE_DIR}"){
+              runParallelTest()
+            }
+          }
+        }
+        /**
+        Build the documentation.
+        */
+        stage('Documentation') {
+          steps {
+            deleteDir()
+            unstash 'source'
+            dir("${BASE_DIR}"){
+              buildDocs(docsDir: "docs", archive: true)
+            }
+          }
+        }
+      }
+    }
+  }
+  post {
+    always {
+      notifyBuildResult()
+    }
+  }
+}
+
+/**
+Parallel task generator for the integration tests.
+*/
+class RumParallelTaskGenerator extends DefaultParallelTaskGenerator {
+
+  public RumParallelTaskGenerator(Map params){
+    super(params)
+  }
+
+  /**
+  build a clousure that launch and agent and execute the corresponding test script,
+  then store the results.
+  */
+  public Closure generateStep(x, y){
+    return {
+      steps.node('linux && immutable && docker'){
+        def label = "${this.tag}:${x}#${y}"
+        try {
+          steps.runScript(label: label, stack: x, scope: y)
+          saveResult(x, y, 1)
+        } catch(e){
+          saveResult(x, y, 0)
+          error("${label} tests failed : ${e.toString()}\n")
+        } finally {
+          steps.junit(allowEmptyResults: true,
+            keepLongStdio: true,
+            testResults: "**/spec/rum-agent-junit.xml")
+          steps.archiveArtifacts(allowEmptyArchive: true, artifacts: "**/.npm/_logs")
+          steps.codecov(repo: 'apm-agent-rum', basedir: "${steps.env.BASE_DIR}",
+            secret: "${steps.env.CODECOV_SECRET}")
+        }
+      }
+    }
+  }
+}
+
+def runParallelTest(){
+  rumTasksGen = new RumParallelTaskGenerator(
```

Comment:
Do we need to duplicate this if we need to add another package? if so maybe we can refactor the common parts into another function.

Reply:
this is because we can not run all test in parallel, we've a limit on Sabucelab, if we need more scopes we can overwrite a method on RumParallelTaskGenerator to do not run in parallel but for now, it does not worth.